pkg/apis/descheduler/v1/types_descheduler.go

```diff
@@ -34,6 +35,10 @@ type KubeDeschedulerSpec struct {
 	// DeschedulingIntervalSeconds is the number of seconds between descheduler runs
 	// +optional
 	DeschedulingIntervalSeconds *int32 `json:"deschedulingIntervalSeconds,omitempty"`
+
+	// PodLifetime is the length of time after which pods should be evicted
+	// This field should be used with profiles that enable the PodLifetime strategy, such as LifecycleAndUtilization
+	PodLifetime *time.Duration `json:"podLifetime,omitempty"`
```

Comment:
I'm trying to think through how to best approach those customizations. This proposed approach imo isn't the best.
I'm thinking of something like:
```go
type KubeDeschedulerSpec struct {
    ...
    // ProfileCustomizations allows to customize default parameters of any given profile
    ProfileCustomizations ProfileCustomizations
}

type ProfileCustomizations struct {
    // AffinityAndTaintsCustomizations allows providing custom values for AffinityAndTains profile
    AffinityAndTaintsCustomizations *AffinityAndTainsCutomizations 
   ...
}
```
Basically, we'd have struct for each profile holding values specific for it. We allow multiple profiles being specified so we need to similarly be able to tweak parameters for each of them distinctively. 
What do you think about that kind of approach? 
@ingvagabund opinions?
@deads2k for API related review/thoughts

Reply:
Putting customization of all strategies under `ProfileCustomizations` makes it more obvious it's a customization.
+1 for this approach. Assuming all profiles that can be configured alongside have disjoint set of strategies so there's clear mapping.

Reply:
I think that may be going to much toward re-writing the upstream API, which we wanted to avoid.

> Assuming all profiles that can be configured alongside have disjoint set of strategies

We already don't have this with `LifecycleAndUtilization` and `DevPreviewLongLifecycle`. I imagine we will probably introduce more overlapping/conflicting profiles in the future, so I think trying to directly map options to profiles could end up with a lot of redundant fields.

Another example is when we add a profile for `HighNodeUtilization`, that profile can't be enabled alongside `LifecycleAndUtilization` (LowNodeUtilization) but it could re-use the same threshold parameters.

So, I agree in that a good compromise would be to put the options under a new field `ProfileCustomizations`, like you suggested, but I do not think they should be another layer further under profile-specific fields

Reply:
Spoke with @damemi on slack, yeah a single `ProfileCustomizations` is the preferred way to go forward.